src/openrct2/ride/vehicle.c

```diff
@@ -1824,6 +1825,20 @@ static void train_ready_to_depart(rct_vehicle* vehicle, uint8 num_peeps_on_train
 	vehicle_invalidate_window(vehicle);
 }
 
+static uint16 ride_get_train_index_from_vehicle(rct_ride* ride, uint16 spriteIndex)
+{
+	uint16 trainIndex = 0;
+	while (ride->vehicles[trainIndex] != spriteIndex)
+	{
+		trainIndex++;
+		if (trainIndex >= ride->num_vehicles || trainIndex >= countof(ride->vehicles))
+		{
+			return VEHICLE_INVALID_ID;
```

Comment:
This is returning an index into ride->vehicles, so its not related to the type of ride->vehicles.